apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -123,6 +129,10 @@ func (api *ProvisionerAPI) getProvisioningInfoBase(m *state.Machine,
 		return result, errors.Trace(err)
 	}
 
+	// Hardcode charm mem constraints to 64MB
+	result.CharmConstraints = constraints.Value{}
+	result.CharmConstraints.Mem = intPtr(64)
```

Comment:
Using `constraints.Value ` isn't suitable because we need to specify separate values for request and limit. And we only are interested in Mem. So I'd just use 2 separate attributes: `CharmMemLimit` and `CharmMemRequest`